Public/Src/Engine/Scheduler/FileMonitoringViolationAnalyzer.cs

```diff
@@ -656,7 +802,12 @@ private Pip TryFindProducer(AbsolutePath producedPath, VersionDisposition versio
             {
                 foreach (FileArtifact fileArtifact in directoryContent)
                 {
-                    RegisterWriteInPathAndUpdateViolations(pip, fileArtifact.Path, reportedViolations);
+                    var outputArtifactInfo = GetOutputMaterializationInfo(pip, outputArtifactsInfo, fileArtifact);
+                    RegisterWriteInPathAndUpdateViolations(pip, fileArtifact.Path, reportedViolations, outputArtifactInfo, out ReportedViolation? allowedSameContentDoubleWriteViolation);
+                    if (allowedSameContentDoubleWriteViolations != null && allowedSameContentDoubleWriteViolation.HasValue)
+                    {

```

Comment:
Is this case allowed?

Reply:
Yeap, good call, this is not needed for exclusive opaques.

---
In reply to: [280218947](https://github.com/Microsoft/BuildXL/pull/224#discussion_r280218947) [](ancestors = 280218947)